### PR TITLE
Fix issue with duplicate IDs on product options, Fix issue #1136 (open correct product tabs)

### PR DESCRIPTION
**Pull request by flair-duncan**
#### What?

In certain circumstances, product option elements end up with the same ID causing a clash, stopping some options being selected.

Each option component has been given a descriptive ID to prevent the clash. As the add to cart action relies on the name attribute, this does not interfere with the cart system.

#### Screenshots

[Video of issue before changes](https://www.useloom.com/share/1e740a49826f44cdbd56f81bd01bd7d3)

[Video after changes](https://www.useloom.com/share/9fe6d969da9445da91532da6f301de23)

#### What?

When the product page URL contains a fragment identifier that relates to tabbed content, i.e. `#tab-similar` or `#tab-warranty`, the correct tab should appear when the page loads.

#### Related Issue

#1136

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 - Add the +/- icons for the category filtering [#1211](https://github.com/bigcommerce/cornerstone/pull/1211)
 - Fix overlapping logo when using "original" sizing and large logos [#1213](https://github.com/bigcommerce/cornerstone/pull/1213)
 - Fix Product Options hiding Add to Cart on a Google AMP page [#1214](https://github.com/bigcommerce/cornerstone/pull/1214)
+- Fix syntactically incorrect code in carousel and bulk-discount-rates components
+- Fix the possibility of duplicate IDs occurring in product options in certain situations
```
Comment (junedkazi) [inline CHANGELOG.md]: Instead of 3 entries. Please move url change into a separate PR. Also just add one changelog entry per PR with link to the PR. So it would look something like this ```
Fix duplicate IDs occurrence in product options in certain situations & syntax fix in carousel and bulk-discount-rates components [#1215](https://github.com/bigcommerce/cornerstone/pull/1215)
```

File: templates/components/products/options/input-checkbox.html [hunk 2]
```diff
@@ -7,17 +7,17 @@
         {{/if}}
     </label>
 
-    <input 
+    <input
         class="form-checkbox"
         type="checkbox"
         name="attribute[{{id}}]"
-        id="attribute-{{id}}"
+        id="attribute_check_{{id}}"
```
Comment (junedkazi) [inline templates/components/products/options/input-checkbox.html]: Can we just stick with using `-` instead on `_`  🍹

**Comment by flair-duncan** [comment]
Hi @junedkazi - changes made as requested, and separated the other issue into PR #1220
